android/source/VideoLocker/src/org/edx/mobile/logger/Logger.java

```diff
@@ -2,10 +2,12 @@
 
 import android.content.Context;
 
+import java.io.Serializable;
+
 /**
  * Created by shahid on 22/1/15.
  */
-public class Logger {
+public class Logger implements Serializable {
```

Comment:
Android tries to serialize the activity during rotation and recreate it after. But unless you force it to, it does not use the java ``Serializable`` protocol to do so. It should save only the data we explicitly tell it to from inside ``Activity#onSaveInstanceState``. Then it gives us an opportunity to restore it either in ``Activity#onCreate`` or  ``Activity#onRestoreInstanceState``

There's some decent discussion here http://stackoverflow.com/questions/10126845/handle-screen-rotation-without-losing-data-android and http://developer.android.com/training/basics/activity-lifecycle/recreating.html

Also, I'm not sure what all the implications are, but we may be better off handling rotation ourselves in some cases. In particular, I suspect that will be friendlier to the video player?